sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/ResourceIdentifier.cs

```diff
@@ -113,7 +119,11 @@ internal static ResourceIdentifier CreateBaseSubscriptionIdentifier(string subsc
                         return CreateBaseResourceGroupIdentifier(subscription, parts[1], parts.Trim(2));
                     case ProvidersKey:
                         {
-                            if (parts.Count > 3)
+                            if (parts.Count == 2 || parts[2] == ProvidersKey)
```

Comment:
same comment as above.  Should be explicit about Microsoft.Features as the namespace for parts[1]